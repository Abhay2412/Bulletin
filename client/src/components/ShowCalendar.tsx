import { useCallback, useState } from "react";
import FullCalendar, {
  DateSelectArg,
  EventApi,
  EventClickArg
} from "@fullcalendar/react";
import dayGridPlugin from "@fullcalendar/daygrid";
import allLocales from "@fullcalendar/core/locales-all";
import interactionPlugin from "@fullcalendar/interaction";
import { INITIAL_EVENTS, createEventId } from "./event-utils";
import "./ShowCalendar.css";

function App() {
  const [currentEvents, setCurrentEvents] = useState<EventApi[]>([]);
  const handleEvents = useCallback(
    (events: EventApi[]) => setCurrentEvents(events),
    []
  );
  const handleDateSelect = useCallback((selectInfo: DateSelectArg) => {
    let description = prompt("Date: " + selectInfo.start + "\n" +  "Please enter the description of the event")?.trim();
    let calendarApi = selectInfo.view.calendar;
    calendarApi.unselect();
    if (description) {
      calendarApi.addEvent({
        id: createEventId(),
        description,
        start: selectInfo.startStr,
        end: selectInfo.endStr,
        allDay: selectInfo.allDay
      });
    }
  }, []);
  const handleEventClick = useCallback((clickInfo: EventClickArg) => {
    if (
      window.confirm(`This event ${clickInfo.event.title}. Do you want to delete?`)
    ) {
      clickInfo.event.remove();
    }
  }, []);
  return (
    <div className="demo-app">
      <div className="demo-app-main">
        <FullCalendar
          plugins={[dayGridPlugin, interactionPlugin]}
          initialView="dayGridMonth"
          selectable={true}
          editable={true}
          initialEvents={INITIAL_EVENTS}
          locales={allLocales}
          locale="eng"
          eventsSet={handleEvents}
          select={handleDateSelect}
          eventClick={handleEventClick}
        />
      </div>
    </div>
  );
}

export default App;
