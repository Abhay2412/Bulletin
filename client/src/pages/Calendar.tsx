import StyledMenu from "../components/StyledMenu";
import ShowCalendar from "../components/ShowCalendar";
import React, { useState } from "react";
import Box from '@mui/material/Box';
import Button from '@mui/material/Button';
import Typography from '@mui/material/Typography';
import Modal from '@mui/material/Modal';
import TextField from "@material-ui/core/TextField";
import MenuItem from '@mui/material/MenuItem';
import FormControl from '@mui/material/FormControl';
import InputLabel from '@mui/material/InputLabel';
import Select, {SelectChangeEvent } from '@mui/material/Select';
import "./Calendar.css";

const style = {
    position: 'absolute' as 'absolute',
    top: '50%',
    left: '50%',
    transform: 'translate(-50%, -50%)',
    width: 700,
    height: 600,
    bgcolor: 'background.paper',
    border: '2px solid #000',
    boxShadow: 24,
    p: 4,
  };

 export default function Calendar () {
    const [open, setOpen] = useState(false);
    const handleOpen = () => setOpen(true);
    const handleClose = () => setOpen(false);
  
    const [textValue, setTextValue] = useState<string>("");
    const handleSave = () => setOpen(false);
    const [tags, setTag] = useState('');

    const handleChange = (event: SelectChangeEvent) => {
        setTag(event.target.value as string);
    };
    
     return (
    <div>
      <a href="/home"><p>Back to Doe Family Board - Main</p></a>
        <div style={{
    position: "relative",
    float: "right",
    bottom: "10px",
    right: "300px"
    }}>
        <StyledMenu/>
        </div>
        <div style={{
    position: "relative",
    float: "right",
    bottom: "10px",
    right: "100px"
    }}>
         <Button style={{
             backgroundColor: "#AA896B",
             fontSize: "14px"
        }}variant="contained" onClick={handleOpen}>+ Add Event</Button>
      <Modal
        open={open}
        onClose={handleClose}
        aria-labelledby="modal-modal-title"
        aria-describedby="modal-modal-description"
      >
        <Box sx={style}>
          <Typography id="modal-modal-description" sx={{ mt: 2 }}>
          <Button style={{
             color: "#AA896B",
             backgroundColor: "#FFFFFF",
             fontSize: "14px",
             left:"600px"
        }}variant="text" disableElevation onClick={handleClose}>Discard</Button>
          <Box
      component="form"
      sx={{
        '& .MuiTextField-root': { m: 1, width: '25ch' },
      }}
      noValidate
      autoComplete="off"
    >
      <div>
      <TextField id="standard-basic" label="Untitled" variant="standard" InputLabelProps={{
    style: { color: "#B8A590"},
  }} InputProps={{ disableUnderline: true }} fullWidth/>
  </div>
     <div>
     <TextField id="standard-basic" label="Date (Month, Day, Year)" variant="standard" InputLabelProps={{
    style: { color: "#675443"},
  }} InputProps={{ disableUnderline: true }} fullWidth/>
      </div>
      <div>
     <TextField id="standard-basic" label="Start Time(24 Hour Format)" variant="standard" InputLabelProps={{
    style: { color: "#675443"},
  }} InputProps={{ disableUnderline: true }} fullWidth/>
      </div>
      <div>
     <TextField id="standard-basic" label="End Time(24 Hour Format)" variant="standard" InputLabelProps={{
    style: { color: "#675443"},
  }} InputProps={{ disableUnderline: true }} fullWidth/>
      </div>
      <div>
      <TextField id="standard-basic" label="Add Description Here" variant="standard" InputLabelProps={{
    style: { color: "#675443"},
  }} InputProps={{ disableUnderline: true }} fullWidth/>
  </div>
  <div>
  <FormControl fullWidth>
  <InputLabel id="demo-simple-select-label">Tags</InputLabel>
  <Select
    labelId="demo-simple-select-label"
    id="demo-simple-select"
    value={tags}
    label="Tags"
    onChange={handleChange}
  >
    <MenuItem value={"appointments"}>Appointments</MenuItem>
    <MenuItem value={"birthdays"}>Birthdays</MenuItem>
    <MenuItem value={"work"}>Work</MenuItem>
    <MenuItem value={"family"}>Family</MenuItem>
    <MenuItem value={"school"}>School</MenuItem>
    <MenuItem value={"activties"}>Activties</MenuItem>
  </Select>
</FormControl>
  </div>
  <div>
  <Button style={{
             color: "#FFFFFF",
             backgroundColor: "#AA896B",
             fontSize: "14px",
             left:"600px",
             top: "110px"
        }}variant="text" disableElevation onClick={handleSave}>Save Event</Button>
  </div>
    </Box>
          </Typography>
        </Box>
      </Modal>
        </div>
        <ShowCalendar/>
    </div>
    
  );
}