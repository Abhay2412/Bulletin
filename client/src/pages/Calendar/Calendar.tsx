import StyledMenu from "../../components/StyledMenu";
import ShowCalendar from "../../components/ShowCalendar";
import React, { useState } from "react";
import Box from "@mui/material/Box";
import Button from "@mui/material/Button";
import Typography from "@mui/material/Typography";
import Modal from "@mui/material/Modal";
import TextField from "@material-ui/core/TextField";
import MenuItem from "@mui/material/MenuItem";
import MenuList from "@mui/material/MenuList";
import FormControl from "@mui/material/FormControl";
import InputLabel from "@mui/material/InputLabel";
import Select, { SelectChangeEvent } from "@mui/material/Select";

import "./Calendar.css";
import { Container } from "@material-ui/core";

const style = {
  position: "absolute",
  top: "50%",
  left: "50%",
  transform: "translate(-50%, -50%)",
  width: 700,
  height: 500,
  overflow: "scroll",
  bgcolor: "background.paper",
  border: "2px solid #000",
  boxShadow: 24,
  p: 4,
};

export default function Calendar() {
  const [open, setOpen] = useState(false);
  const handleOpen = () => setOpen(true);
  const handleClose = () => setOpen(false);

  const [textValue, setTextValue] = useState<string>("");
  const handleSave = () => setOpen(false);
  const [tags, setTag] = useState("");

  const handleChange = (event: SelectChangeEvent) => {
    setTag(event.target.value as string);
  };

  return (
    <Container>
      <div
        style={{
          display: "flex",
          flexDirection: "row",
          justifyContent: "space-between",
          margin: "30px auto 5px auto",
          width: "90%",
        }}
      >
        <a href="/home">
          <p>Back to Doe Family Board - Main</p>
        </a>
        <div
          style={{
            display: "flex",
            flexDirection: "row",
          }}
        >
          <div>
            <StyledMenu />
          </div>
          <Button
            style={{
              backgroundColor: "#AA896B",
              fontSize: "14px",
            }}
            variant="contained"
            onClick={handleOpen}
          >
            + Add Event
          </Button>
        </div>
      </div>
      <div>
        <Modal
          open={open}
          onClose={handleClose}
          aria-labelledby="modal-modal-title"
          aria-describedby="modal-modal-description"
        >
          <Box sx={style}>
            <Typography id="modal-modal-description" sx={{ mt: 2 }}>
              <Button
                style={{
                  color: "#AA896B",
                  backgroundColor: "#FFFFFF",
                  fontSize: "14px",
                  left: "600px",
                }}
                variant="text"
                disableElevation
                onClick={handleClose}
              >
                Discard
              </Button>
              <Box
                component="form"
                sx={{
                  "& .MuiTextField-root": { m: 1, width: "25ch" },
                }}
                noValidate
                autoComplete="off"
              >
                <div>
                  <TextField
                    id="standard-basic"
                    label="Name"
                    variant="standard"
                    InputLabelProps={{
                      style: { color: "#B8A590" },
                    }}
                    InputProps={{ disableUnderline: true }}
                    fullWidth
                  />
                </div>
                <div>
                  <TextField
                    id="standard-basic"
                    label="Date (Month, Day, Year)"
                    variant="standard"
                    InputLabelProps={{
                      style: { color: "#675443" },
                    }}
                    InputProps={{ disableUnderline: true }}
                    fullWidth
                  />
                </div>
                <div>
                  <TextField
                    id="standard-basic"
                    label="Add Description Here"
                    variant="standard"
                    InputLabelProps={{
                      style: { color: "#675443" },
                    }}
                    InputProps={{ disableUnderline: true }}
                    fullWidth
                  />
                </div>
                <div>
                  <FormControl fullWidth>
                    <InputLabel id="demo-simple-select-label">Tags</InputLabel>
                    <Select
                      labelId="demo-simple-select-label"
                      id="demo-simple-select"
                      value={tags}
                      label="Tags"
                      onChange={handleChange}
                    >
                      <MenuItem value={"appointments"}>Appointments</MenuItem>
                      <MenuItem value={"birthdays"}>Birthdays</MenuItem>
                      <MenuItem value={"work"}>Work</MenuItem>
                      <MenuItem value={"family"}>Family</MenuItem>
                      <MenuItem value={"school"}>School</MenuItem>
                      <MenuItem value={"activties"}>Activties</MenuItem>
                    </Select>
                  </FormControl>
                </div>
                <div>
                  <Button
                    style={{
                      color: "#FFFFFF",
                      backgroundColor: "#AA896B",
                      fontSize: "14px",
                      left: "600px",
                      top: "110px",
                    }}
                    variant="text"
                    disableElevation
                    onClick={handleSave}
                  >
                    Save Event
                  </Button>
                </div>
              </Box>
            </Typography>
          </Box>
        </Modal>
      </div>
      <ShowCalendar />
    </Container>
  );
}
