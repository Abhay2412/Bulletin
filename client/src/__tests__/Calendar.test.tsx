import React from "react";
import { configure, mount, ReactWrapper, shallow } from "enzyme";
import ShowCalendar from "../components/ShowCalendar";
import Calendar from "../pages/Calendar/Calendar";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import renderer from "react-test-renderer";
import { store } from "../store";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";

configure({ adapter: new Adapter() });

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useParams: () => ({
    board_id: "g7HGuvfnGe3nhQgAF8qY",
  }),
}));

describe("testing <Calendar /> component", () => {
  const wrapper: ReactWrapper = mount(
    <Provider store={store}>
      <BrowserRouter>
        <Calendar />
      </BrowserRouter>
    </Provider>
  );

  it("should render calendar properly", () => {
    expect(wrapper.getElements()).toMatchSnapshot();
  });

  it("should render ShowCalendar properly", () => {
    const showCalendar = mount(
      <ShowCalendar
        events={[
          {
            id: "NWbV0lZqOAo7q8CzwUye",
            title: "My Birthday",
            start: "2022-03-28T01:00:00",
            end: "2022-03-28T02:00:00",
          },
        ]}
        setEvents={() => {}}
      />
    );
    expect(showCalendar.getElements()).toMatchSnapshot();
  });

  it("should show window alert", () => {
    const showCalendar = mount(
      <ShowCalendar
        events={[
          {
            id: "NWbV0lZqOAo7q8CzwUye",
            title: "My Birthday",
            start: "2022-03-28T01:00:00",
            end: "2022-03-28T02:00:00",
          },
        ]}
        setEvents={() => {}}
      />
    );

    showCalendar.find("a").find(".fc-daygrid-event").simulate("click");
    expect(window.confirm).toBeTruthy();
  });

  it("should open a modal", () => {
    wrapper.find("button").find("#calendar-add-event-button").simulate("click");
    expect(wrapper.find("#calendar-add-event-modal").exists()).toBeTruthy();
  });

  describe("should have add event modal open", () => {
    beforeAll(() => {
      wrapper
        .find("button")
        .find("#calendar-add-event-button")
        .simulate("click");
    });

    it("should change event name field value", () => {
      wrapper
        .find("input")
        .find("#calendar-add-event-modal-event-name")
        .simulate("change", {
          target: { value: "Test Event" },
        });
      expect(
        wrapper
          .find("input")
          .find("#calendar-add-event-modal-event-name")
          .get(0).props.value
      ).toEqual("Test Event");
    });

    it("should show event name error", () => {
      wrapper
        .find("input")
        .find("#calendar-add-event-modal-event-name")
        .simulate("change", {
          target: { value: "" },
        });

      wrapper
        .find("button")
        .find("#calendar-add-event-modal-save-button")
        .simulate("click");
      expect(
        wrapper
          .find("p")
          .find("#calendar-add-event-modal-event-name-helper-text")
          .text()
      ).toEqual("Enter an event name.");
    });

    it("should change event date field value", () => {
      wrapper
        .find("input")
        .find("#calendar-add-event-modal-event-date")
        .simulate("change", {
          target: { value: "03,29,2022" },
        });
      expect(
        wrapper
          .find("input")
          .find("#calendar-add-event-modal-event-date")
          .get(0).props.value
      ).toEqual("03,29,2022");
    });

    it("should show event date empty error", () => {
      wrapper
        .find("input")
        .find("#calendar-add-event-modal-event-date")
        .simulate("change", {
          target: { value: "" },
        });

      wrapper
        .find("button")
        .find("#calendar-add-event-modal-save-button")
        .simulate("click");
      expect(
        wrapper
          .find("p")
          .find("#calendar-add-event-modal-event-date-helper-text")
          .text()
      ).toEqual("Enter a date.");
    });

    it("should show invalid event date error", () => {
      wrapper
        .find("input")
        .find("#calendar-add-event-modal-event-date")
        .simulate("change", {
          target: { value: "asdzs" },
        });

      wrapper
        .find("button")
        .find("#calendar-add-event-modal-save-button")
        .simulate("click");
      expect(
        wrapper
          .find("p")
          .find("#calendar-add-event-modal-event-date-helper-text")
          .text()
      ).toEqual("Enter date in the format MM,DD,YYYY");
    });

    it("should change event start time value", () => {
      wrapper
        .find("input")
        .find("#calendar-add-event-modal-event-start-time")
        .simulate("change", {
          target: { value: "01:00:00" },
        });
      expect(
        wrapper
          .find("input")
          .find("#calendar-add-event-modal-event-start-time")
          .get(0).props.value
      ).toEqual("01:00:00");
    });

    it("should show empty event start time error", () => {
      wrapper
        .find("input")
        .find("#calendar-add-event-modal-event-start-time")
        .simulate("change", {
          target: { value: "" },
        });

      wrapper
        .find("button")
        .find("#calendar-add-event-modal-save-button")
        .simulate("click");
      expect(
        wrapper
          .find("p")
          .find("#calendar-add-event-modal-event-start-time-helper-text")
          .text()
      ).toEqual("Enter a start time.");
    });

    it("should show invalid event start time error", () => {
      wrapper
        .find("input")
        .find("#calendar-add-event-modal-event-start-time")
        .simulate("change", {
          target: { value: "invalid" },
        });

      wrapper
        .find("button")
        .find("#calendar-add-event-modal-save-button")
        .simulate("click");
      expect(
        wrapper
          .find("p")
          .find("#calendar-add-event-modal-event-start-time-helper-text")
          .text()
      ).toEqual("Enter time in the format HH:MM:SS");
    });
  });
});
