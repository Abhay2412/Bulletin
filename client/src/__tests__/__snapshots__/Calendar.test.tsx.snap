// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing <Calendar /> component should render ShowCalendar properly 1`] = `
Array [
  <ShowCalendar
    events={
      Array [
        Object {
          "end": "2022-03-28T02:00:00",
          "id": "NWbV0lZqOAo7q8CzwUye",
          "start": "2022-03-28T01:00:00",
          "title": "My Birthday",
        },
      ]
    }
    setEvents={[Function]}
  />,
]
`;

exports[`testing <Calendar /> component should render calendar properly 1`] = `
Array [
  <Provider
    store={
      Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <BrowserRouter>
      <Calendar />
    </BrowserRouter>
  </Provider>,
]
`;
